### PR TITLE
ci: push build images to Docker hub

### PR DESCRIPTION
**Pull request by subhamkrai**
push build image to Dockerhub every time when
there is a push in the master branch using GH actions.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: .github/workflows/push-build.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      -
```
Comment (leseb) [inline .github/workflows/push-build.yaml:12]: wrong placement of the `-`.

File: .github/workflows/push-build.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (leseb) [inline .github/workflows/push-build.yaml:14]: Why do we need QEMU?

File: .github/workflows/push-build.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      -
+        name: Build and push
```
Comment (leseb) [inline .github/workflows/push-build.yaml:25]: Where do you build Rook and how is it pushed?

**Review by leseb — CHANGES_REQUESTED** [review]

File: .github/workflows/push-build.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: Done.

File: .github/workflows/push-build.yaml (hunk 2 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: Changed the workflow. Done.

File: .github/workflows/push-build.yaml (hunk 3 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: Changed the workflow. Done.

File: .github/workflows/push-build.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+jobs:
+  image-build:
```
Comment (leseb) [inline .github/workflows/push-build.yaml:7]:
```suggestion
  push-master-image-on-docker:
```

File: .github/workflows/push-build.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        go-version: 1.15
+
+    - name: build rook
+      run: |
+        # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+        GOPATH=$(go env GOPATH) make clean && make -j$nproc VERSION=0 build
+        docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
+        docker images
+        docker tag $(docker images|awk '/build-/ {print $1}') rook/ceph:latest
+        docker push rook/ceph:latest
```
Comment (travisn) [inline .github/workflows/push-build.yaml:25]: Each master build currently pushes two tags: `rook/ceph:master` and something like `rook/ceph:v1.5.0.80.gaac0e5f` based on a git command. We don't use the `latest` tag. See the tags published [here](https://hub.docker.com/r/rook/ceph/tags/). We also push both amd64 and arm64 images, so this task will be a bit more complicated.

**Review by travisn — CHANGES_REQUESTED** [review]
Marking changes requested per earlier comment

File: .github/workflows/push-build.yaml (hunk 5 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml:25]: 1. First task can be using `${{ secrets.DOCKER_USER }}/back:${{ github.sha }}`?
2. for amd64, arm64 see this https://github.com/marketplace/actions/build-and-push-docker-images. I'm trying to understand how I can use this.

**Comment by travisn** [comment]
Something else the publish action currently does is to update the published documentation, so we will also need to include that with this action. See [this makefile](https://github.com/rook/rook/blob/master/build/release/Makefile#L119-L131)

**Comment by subhamkrai** [comment]
The changes are not perfect yet I need to validate and understand a few things.

File: .github/workflows/push-build.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+jobs:
+  push-master-image-on-docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Prepare
+        run: |
+          DOCKER_IMAGE=${{ secrets.DOCKER_USERNAME }}/${GITHUB_REPOSITORY#*/}
```
Comment (leseb) [inline .github/workflows/push-build.yaml:15]: Let's write bash scripts instead

File: .github/workflows/push-build.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        env:
+          BRANCH_NAME : $(git branch | grep \* | cut -d ' ' -f2)
+          DOCS_VERSION : $(echo ${BRANCH_NAME} | sed -E "s/^release\-([0-9]+)\.([0-9]+)$$/v\1.\2/g")
+          MY_SECRET : ${{ secrets.GITHUB_TOKEN }}
+          DOCS_DIR : Documentation
+          DOCS_WORK_DIR : https://rook.github.io
+          DOCS_VERSION_DIR : $(DOCS_WORK_DIR)/docs/rook/$(DOCS_VERSION)
+        run: |
+          
+          if [[ $MY_SECRET != ""]]; then
```
Comment (leseb) [inline .github/workflows/push-build.yaml:66]: ditto

**Review by leseb — CHANGES_REQUESTED** [review]

File: .github/workflows/push-build.yaml (hunk 6 above)
Comment (travisn) [reply .github/workflows/push-build.yaml:15]: Actually, we already have makefile commands, can we just use those?

File: .github/workflows/push-build.yaml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+          rm -rf $(DOCS_WORK_DIR)
+          mkdir -p $(DOCS_WORK_DIR)
+          git clone --depth=1 -b master $(DOCS_GIT_REPO) $(DOCS_WORK_DIR)
+          rm -rf $(DOCS_VERSION_DIR)
+          cp -r $(DOCS_DIR)/ $(DOCS_VERSION_DIR)
+          cd $(DOCS_WORK_DIR) && DOCS_VERSION=$(DOCS_VERSION) $(MAKE) _data/projects.json
+          cd $(DOCS_WORK_DIR) && DOCS_VERSION=$(DOCS_VERSION) $(MAKE) publish
+  
+
```
Comment (BlaineEXE) [inline .github/workflows/push-build.yaml:80]: Nit: clean up whitespace at end of file

File: .github/workflows/push-build.yaml (hunk 4 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: Done.

**Comment by subhamkrai** [comment]
this PR is still on progress need some testing. @travisn helping me on testing.
Thanks

**Comment by mykaul** [comment]
With Dockerhub limiting IO, why are we not pushing to Quay.IO?

**Comment by leseb** [comment]
> With Dockerhub limiting IO, why are we not pushing to Quay.IO?

We raised the subject in the community meeting end of last year, somehow we haven't got any CI or end-user issues.
Rook's CI always builds the container image locally (so no pulling from the Docker hub) and we are fetching Ceph images from https://hub.docker.com/r/ceph/ where the Ceph org has a "Team Plan" with unlimited pulls. So I guess this helps.

**Comment by travisn** [comment]
> this PR is still on progress need some testing. @travisn helping me on testing.
> Thanks

Remind me what is remaining here? You're asking for independent testing of the action?

**Comment by subhamkrai** [comment]
> > this PR is still on progress need some testing. @travisn helping me on testing.
> > Thanks
> 
> Remind me what is remaining here? You're asking for independent testing of the action?

Currently,
1. When I push from `master branch` I see 2 tags `master` and `0f57224(only git sha id...)`

So, Remaining things need to be tested which I can think of
1. need to test with tags
2. Need to verify the changes for the new release or testing from `release branch`
3. Need to verify if published  documentation is updated

JFYI, I'm using Makefile commands so above things should work as Jenkins uses same commands for the same.

**Comment by ibotty** [comment]
Note:  I run into docker limits regularly.  For now I am always mirroring the image locally which I'd like to avoid.

**Comment by leseb** [comment]
@subhamkrai any updates?

**Comment by subhamkrai** [comment]
> @subhamkrai any updates?

AFAIR last time, the master push to the docker hub was successful... was discussing this @travisn  about the release process...will back to this soon.

File: .github/workflows/push-build.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+name: Build Image
+on:
+  push:
+    branches:
+      - master
+      - 'releases/**'
```
Comment (travisn) [inline .github/workflows/push-build.yaml:6]: Let's remove the release branch and tags triggers, and just do this on the master branch for now. In a later PR we can enable for the others after we are sure master is stable with this.

File: .github/workflows/push-build.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    tags:
+      - v*
+jobs:
+  push-master-image-on-docker:
```
Comment (travisn) [inline .github/workflows/push-build.yaml:10]:
```suggestion
  push-image-to-docker:
```

File: tests/scripts/build-release.sh [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    build/run make -j$(nproc) -C build/release build BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_PSW}
+    git status &
+    git diff &
+    build/run make -j$(nproc) -C build/release publish BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_PSW}
+}
+
+
+
+
+if [ $# -ge 1 ] && echo "$1" | grep "relese-" ; then
```
Comment (travisn) [inline tests/scripts/build-release.sh:35]:
```suggestion
if [ $# -ge 1 ] && echo "$1" | grep "release-" ; then
```

File: tests/scripts/build-release.sh [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+function release_branch() {
+    build/run make -j$(nproc) build.all
+    build/run make -j$(nproc) mod.check
+    build/run make -j$(nproc) -C build/release build BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_PSW}
+    git status &
+    git diff &
+    build/run make -j$(nproc) -C build/release publish BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_PSW}
+}
+
+
```
Comment (travisn) [inline tests/scripts/build-release.sh:32]: nit: remove extra lines

File: tests/scripts/build-release.sh [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+# VARIABLES #
+#############
+BRANCH_NAME=$1
+
+#############
+# FUNCTIONS #
+#############
+
+function  master_branch() {
+    build/run make -j$(nproc) VERSION=0 build.all
```
Comment (travisn) [inline tests/scripts/build-release.sh:14]: If this step fails, it aborts the rest of the script, right? Of course we can't push if the build fails

File: tests/scripts/build-release.sh [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    build/run make -j$(nproc) mod.check
+    build/run make -j$(nproc) -C build/release build BRANCH_NAME=${BRANCH_NAME}  GIT_API_TOKEN=${GIT_PSW}
+    git status &
+    git diff &
+    build/run make -j$(nproc) -C build/release publish BRANCH_NAME=${BRANCH_NAME} GIT_API_TOKEN=${GIT_PSW}
+    build/run make -j$(nproc) -C build/release promote TAG_WITH_SUFFIX=true BRANCH_NAME=master CHANNEL=master
+}
+
+function release_branch() {
```
Comment (travisn) [inline tests/scripts/build-release.sh:23]: Can we combine the master and release functions? Inside the single function we can have conditional commands where needed

File: tests/scripts/build-release.sh [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+#############
+
+function  master_branch() {
+    build/run make -j$(nproc) VERSION=0 build.all
+    build/run make -j$(nproc) mod.check
+    build/run make -j$(nproc) -C build/release build BRANCH_NAME=${BRANCH_NAME}  GIT_API_TOKEN=${GIT_PSW}
+    git status &
+    git diff &
+    build/run make -j$(nproc) -C build/release publish BRANCH_NAME=${BRANCH_NAME} GIT_API_TOKEN=${GIT_PSW}
+    build/run make -j$(nproc) -C build/release promote TAG_WITH_SUFFIX=true BRANCH_NAME=master CHANNEL=master
```
Comment (travisn) [inline tests/scripts/build-release.sh:20]: - Promote really only publishes the helm chart. It would be nice to rename this action so it's more clear, but that could be a separate PR.
- We should automatically `promote` 1) every master branch and 2) every tag. But we should *not* promote the release branch builds (unless tagged).

File: build/release/Makefile [hunk 16]
```diff
@@ -61,7 +61,8 @@ FLAVORS ?= output
 override BRANCH_NAME := pr/$(BRANCH_NAME)
 endif
 
-DOCKER_REGISTRY ?= rook
+# DOCKER_REGISTRY ?= rook
```
Comment (travisn) [inline build/release/Makefile:64]: All changes in this makefile will be reverted before merge, right? Looks like they are all for temporary testing.

File: tests/scripts/build-release.sh (hunk 13 above)
Comment (travisn) [reply tests/scripts/build-release.sh:14]: Why VERSION=0?

**Review by travisn — CHANGES_REQUESTED** [review]

**Comment by travisn** [comment]
To capture one more issue we discussed... With this PR let's also delete the "Publish if Master" step in the Jenkinsfile so we can verify this change after the merge.

File: .github/workflows/push-build.yaml (hunk 10 above)
Comment (leseb) [inline .github/workflows/push-build.yaml:10]:
```suggestion
  push-image-to-container-registry:
```

Let's make it generic.

File: .github/workflows/push-build.yaml [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
```
Comment (leseb) [inline .github/workflows/push-build.yaml:11]:
```suggestion
    runs-on: 18.04
```

Let's stay consistent with what we have today

File: .github/workflows/push-build.yaml [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
```
Comment (leseb) [inline .github/workflows/push-build.yaml:17]: Why?

File: .github/workflows/push-build.yaml [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+    branches:
+      - master
+      - 'releases/**'
+    tags:
+      - v*
+jobs:
+  push-master-image-on-docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
```
Comment (leseb) [inline .github/workflows/push-build.yaml:13]:
```suggestion
      - name: checkout
```

File: .github/workflows/push-build.yaml [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
+
+      - name: Login to DockerHub
```
Comment (leseb) [inline .github/workflows/push-build.yaml:21]:
```suggestion
      - name: login to container registry
```

File: .github/workflows/push-build.yaml [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: build and release
+        env:
```
Comment (leseb) [inline .github/workflows/push-build.yaml:28]: env:
        GIT_PSW: ${{ secrets.GIT_PSW }}
        BRANCH_NAME: ${{ github.base_ref }}

Let's use this instead of `${GITHUB_REF#refs/heads/}`

File: tests/scripts/build-release.sh [hunk 22]
```diff
@@ -0,0 +1,39 @@
+/bin/bash -e
```
Comment (leseb) [inline tests/scripts/build-release.sh:1]: broken shebang and use env bash instead

File: tests/scripts/build-release.sh (hunk 13 above)
Comment (leseb) [reply tests/scripts/build-release.sh:14]: As mentioned earlier we need to fix this before pushing this. We use `VERSION=0` because the helm build was breaking. Now we need to fix this instead of the workaround.

**Review by leseb — CHANGES_REQUESTED** [review]

File: .github/workflows/push-build.yaml (hunk 18 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml:17]: docker/setup-qemu action installs QEMU static binaries, which are used to run builders for architectures other than the host.`

File: .github/workflows/push-build.yaml (hunk 20 above)
Comment (BlaineEXE) [reply .github/workflows/push-build.yaml]: Nit: `log in`

File: .github/workflows/push-build.yaml (hunk 10 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: done.

File: tests/scripts/build-release.sh (hunk 22 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh]: yeah, it was missed, changed now.

File: tests/scripts/build-release.sh (hunk 14 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh]: done.

File: tests/scripts/build-release.sh (hunk 12 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh]: removed.

File: .github/workflows/push-build.yaml (hunk 9 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: done.

File: .github/workflows/push-build.yaml (hunk 21 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml:28]: `github.base_ref` is only for `pull_request event` this is not working...show I have to make tweak a little to work

File: .github/workflows/push-build.yaml (hunk 20 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: done.

File: tests/scripts/build-release.sh [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+# FUNCTIONS #
+#############
+
+function  build() {
+    # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+    build/run make VERSION=0 build.all
+    build/run make mod.check
+}
+
+if echo "${BRANCH_NAME}" | grep "release-" ; then
```
Comment (travisn) [inline tests/scripts/build-release.sh:14]: The steps for master and release branches should really be the same. The only difference I can think of is that a non-tagged release branch would *not* run the promote action (for the helm chart). So we should be able to combine these steps, and just have a condition only on the promote.

File: build/release/Makefile [hunk 24]
```diff
@@ -138,20 +138,20 @@ HELM_URL := $(HELM_BASE_URL)/$(CHANNEL)
 
 promote.helm: $(HELM)
 #	copy existing charts to a temp dir, then combine with new charts, reindex, and upload
-	@$(S3_SYNC) s3://$(HELM_S3_BUCKET)/$(CHANNEL) $(HELM_TEMP)
-	@$(S3_SYNC) s3://$(S3_BUCKET)/build/$(BRANCH_NAME)/$(VERSION)/charts $(HELM_TEMP)
+#	@$(S3_SYNC) s3://$(HELM_S3_BUCKET)/$(CHANNEL) $(HELM_TEMP)
+#	@$(S3_SYNC) s3://$(S3_BUCKET)/build/$(BRANCH_NAME)/$(VERSION)/charts $(HELM_TEMP)
```
Comment (travisn) [inline build/release/Makefile:142]: We still need to push the helm chart and artifacts to s3 for master and release builds, i'm still not clear why this would remain out.

**Review by travisn — CHANGES_REQUESTED** [review]

File: tests/scripts/build-release.sh (hunk 23 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh:14]: we have `TAG_WITH_SUFFIX=true` for release so I think we can't combine anymore.

for `release`
`build/run make -C build/release build BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_PSW}`

for `master`

` build/run make -C build/release build BRANCH_NAME=${BRANCH_NAME}  GIT_API_TOKEN=${GIT_PSW}`

File: build/release/Makefile (hunk 24 above)
Comment (subhamkrai) [reply build/release/Makefile]: I have reverted the changes done in Makefile. Thanks

File: .github/workflows/push-build.yaml [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+      # creating custom env var
+      - name: set env
+        run: echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: build and release
+        env:
+          GIT_PSW: ${{ secrets.GIT_PSW }}
+          BRANCH_NAME: ${{ env.BRANCH_NAME }}
+          AWS_USR: ${{ secrets.AWS_USR }}
+          AWS_PSW: ${{ secrets.AWS_PSW }}
```
Comment (subhamkrai) [inline .github/workflows/push-build.yaml:40]:
```suggestion
          AWS_USR: ${{ secrets.AWS_USR }}
          AWS_PSW: ${{ secrets.AWS_PSW }}
```

@travisn @leseb please confirm if this is correct naming. Thanks

**Comment by subhamkrai** [comment]
FYI, I'm not able to test after adding AWS credential.

File: tests/scripts/build-release.sh (hunk 23 above)
Comment (travisn) [reply tests/scripts/build-release.sh:15]: Got it, I overlooked the TAG_WITH_SUFFIX

File: .github/workflows/push-build.yaml [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+        run: tests/scripts/build-release.sh
```
Comment (BlaineEXE) [inline .github/workflows/push-build.yaml:41]: Nit: newline at end of file

File: tests/scripts/build-release.sh [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+    build/run make -C build/release publish BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_PSW}
+elif echo "${BRANCH_NAME}" | grep "master\|tag" ; then
+    build
+    build/run make -C build/release build BRANCH_NAME=${BRANCH_NAME}  GIT_API_TOKEN=${GIT_PSW}
+    git status &
+    git diff &
+    build/run make -C build/release publish BRANCH_NAME=${BRANCH_NAME} AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_PSW}
+    # automatically promote the master builds
+    build/run make -C build/release promote BRANCH_NAME=master CHANNEL=master AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW}           
+fi
```
Comment (BlaineEXE) [inline tests/scripts/build-release.sh:29]: Nit: newline at end of file

File: .github/workflows/push-build.yaml (hunk 26 above)
Comment (BlaineEXE) [reply .github/workflows/push-build.yaml:41]: FWIW, if you use VSCode, there is an option to ensure a newline at the end of files on saving.

File: .github/workflows/push-build.yaml (hunk 25 above)
Comment (leseb) [reply .github/workflows/push-build.yaml:40]: They are not set in our action Secret env so they can be whatever you want. I don't have access to these credential. @travisn please add them to list of Secrets env in Rook. Thanks!

File: .github/workflows/push-build.yaml (hunk 26 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: added.

File: .github/workflows/push-build.yaml (hunk 25 above)
Comment (travisn) [reply .github/workflows/push-build.yaml:40]: Ok, I added the two secrets to the github settings!

**Review by travisn — APPROVED** [review]

**Review by travisn — APPROVED** [review]

File: tests/scripts/build-release.sh [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+    # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+    build/run make VERSION=0 build.all
+    # quick check that go modules are tidied
+    build/run make mod.check
+}
+
+if echo "${BRANCH_NAME}" | grep "release-" ; then
+    build
+    build/run make -C build/release build BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_PSW}
+    git status &
```
Comment (leseb) [inline tests/scripts/build-release.sh:18]: Why do you run the command in the background?

File: tests/scripts/build-release.sh [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+    git diff &
+    build/run make -C build/release publish BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_PSW}
+elif echo "${BRANCH_NAME}" | grep "master\|tag" ; then
```
Comment (leseb) [inline tests/scripts/build-release.sh:21]:
```suggestion
elif echo "${BRANCH_NAME}" | grep -E "master|tag" ; then
```

Are you trying to match one of two?

**Review by leseb — CHANGES_REQUESTED** [review]

File: tests/scripts/build-release.sh (hunk 28 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh:18]: When I was testing... screen holding/stuck in this(don't remember the exact problem) but there was something.

File: tests/scripts/build-release.sh (hunk 29 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh:21]: Yes...

Comment (leseb) [reply tests/scripts/build-release.sh:21]: Actually go with:

```sh
elif [[ ${BRANCH_NAME} =~ master|tag ]]; then
```

File: tests/scripts/build-release.sh [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+#############
+
+function  build() {
+    # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+    build/run make VERSION=0 build.all
+    # quick check that go modules are tidied
+    build/run make mod.check
+}
+
+if echo "${BRANCH_NAME}" | grep "release-" ; then
```
Comment (leseb) [inline tests/scripts/build-release.sh:15]:
```suggestion
if [[ ${BRANCH_NAME} =~ release- ]]; then
```

**Review by leseb — CHANGES_REQUESTED** [review]

File: tests/scripts/build-release.sh (hunk 29 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh]: done.

File: tests/scripts/build-release.sh (hunk 30 above)
Comment (subhamkrai) [reply tests/scripts/build-release.sh]: done

File: .github/workflows/push-build.yaml [hunk 31]
```diff
@@ -0,0 +1,41 @@
+name: Build Image
```
Comment (leseb) [inline .github/workflows/push-build.yaml:1]:
```suggestion
name: Push Image Build
```

**Review by leseb — CHANGES_REQUESTED** [review]
small nit

File: .github/workflows/push-build.yaml (hunk 31 above)
Comment (subhamkrai) [reply .github/workflows/push-build.yaml]: done

File: tests/scripts/build-release.sh (hunk 28 above)
Comment (leseb) [reply tests/scripts/build-release.sh:18]: Did you try removing this? This shouldn't be necessary.

Comment (subhamkrai) [reply tests/scripts/build-release.sh:18]: No, because anyway we need this later for the release branch...so I kept it.

Comment (leseb) [reply tests/scripts/build-release.sh:18]: What do you "we were doing it?" I don't see it in the Jenkins file

Comment (subhamkrai) [reply tests/scripts/build-release.sh:18]: So this is for checking master and release. We can remove this for now as this PR is all about master but later we need this when doing for release.

Comment (leseb) [reply tests/scripts/build-release.sh:18]: Let's revisit this after this merge.

**Review by leseb — APPROVED** [review]
